Classes/Domain/Model/StructuredJobDefinitionTrait.php

```diff
@@ -0,0 +1,670 @@
+            'jobBenefits' => $this->getJobBenefits(),
+            'jobLocation' => $this->getStructuredLocationData(),
+            'occupationalCategory' => $this->getOccupationalCategory(),
+            'qualifications' => $this->getQualifications(),
+            'responsibilities' => $this->getResponsibilities(),
+            'skills' => $this->getSkills(),
+            'specialCommitments' => $this->getSpecialCommitments(),
+            'workHours' => $this->getWorkHours(),
+        ];
+
+        if ($this->getEndtime() != null) {
+            $data['validThrough'] = $this->getEndtime()->format('Y-m-d');
+        }
+
+        return $data;
+    }
+
+    /**
+     * Returns the hiring organization in structured form
+     *
+     * @return array
+     */
+    public function getStructuredHiringOrganization()
+    {
+        if ($this->overrideGlobalHiringOrganization) {
+            $data = [
+                '@type' => 'Organization',
+                'name' => $this->getHiringOrganization(),
+            ];
+        } else {
+            $settings = TyposcriptService::getInstance()->getSettings();
+            $data = [
+                '@type' => 'Organization',
+                'name' => $settings['companyData']['name'],
+            ];
+        }
+        return $data;
+    }
+
+    /**
+     * Returns job location data in structured form
+     *
+     * @return array
+     */
+    public function getStructuredLocationData()
+    {
+        if ($this->overrideGlobalLocation) {
+            $data = [
+                '@type' => 'Place',
+                'address' => [
+                    'streetAddress' => $this->getJobLocationAddressStreetAddress(),
+                    'addressLocality' => $this->getJobLocationAddressLocality(),
+                    'addressRegion' => $this->getJobLocationAddressRegion() ? $this->getJobLocationAddressRegion()->getLocalName() : '',
+                    'postalCode' => $this->getJobLocationAddressPostalCode(),
+                    'addressCountry' => $this->getJobLocationAddressCountry() ? $this->getJobLocationAddressCountry()->getIsoCodeA3() : '',
+                ],
+            ];
+        } else {
+            $settings = TyposcriptService::getInstance()->getSettings();
+            $data = [
+                '@type' => 'Place',
+                'address' => [
+                    'streetAddress' => $settings['companyData']['street'],
+                    'addressLocality' => $settings['companyData']['locality'],
+                    'addressRegion' => $settings['companyData']['locality'],
+                    'postalCode' => $settings['companyData']['postalCode'],
+                    'addressCountry' => $settings['companyData']['country'],
+                ],
+            ];
+        }
+        return $data;
+    }
+
+    /**
+     * @var string $jobTitle
+     */
+    protected $jobTitle;
+
+    /**
+     * @return string
+     */
+    public function getJobTitle()
```

Comment:
I think you need neither getter nor setter here, the property should be enough. Extbase will inject the values via reflection anyways and you don't risk adding adding ways to modify model objects you didn't intend.

Reply:
Removed, they will be re-added in a separate PR.